cpp/test/cwrapper/cwrapper_test.cc

```diff
@@ -28,23 +39,145 @@ extern "C" {
+    text_measurement.encoding = TS_ENCODING_PLAIN;
+
+    timeseries_schema date_measurement;
+    date_measurement.timeseries_name = date_measurement_id;
+    date_measurement.compression = TS_COMPRESSION_UNCOMPRESSED;
+    date_measurement.data_type = TS_DATATYPE_DATE;
+    date_measurement.encoding = TS_ENCODING_PLAIN;
+
+    // Create TsFile writer
+    auto* writer = (storage::TsFileWriter*)_tsfile_writer_new(
+        filename, 128 * 1024 * 1024, &code);
+    ASSERT_OK(code, "create writer failed");
+
+    // Register time series with the writer
+    ASSERT_OK(
+        _tsfile_writer_register_timeseries(writer, device_id, &str_measurement),
+        "register timeseries failed");
+
+    ASSERT_OK(_tsfile_writer_register_timeseries(writer, device_id,
+                                                 &text_measurement),
+              "register timeseries failed");
+
+    ASSERT_OK(_tsfile_writer_register_timeseries(writer, device_id,
+                                                 &date_measurement),
+              "register timeseries failed");
+
+    // Create a new time series record
+    auto* record = _ts_record_new(device_id, 0, 3);
+
+    // Insert string data
+    char* test_str = "test_string";
+    ASSERT_OK(_insert_data_into_ts_record_by_name_string(
+                  record, str_measurement_id, test_str),
+              "insert data failed");
+
+    // Insert text data
+    char* test_text = "test_text";
+    ASSERT_OK(_insert_data_into_ts_record_by_name_string(
+                  record, text_measurement_id, test_text),
+              "insert data failed");
+
+    // Insert date data - NOTE: There's a bug here, should use
+    // date_measurement_id
+    int32_t test_date = 20251118;
+    ASSERT_OK(_insert_data_into_ts_record_by_name_int32_t(
+                  record, date_measurement_id, test_date),
+              "insert data failed");
+
+    // Write the record to file and close writer
+    ASSERT_OK(_tsfile_writer_write_ts_record(writer, record),
+              "write record failed");
+    ASSERT_OK(_tsfile_writer_flush(writer), "flush failed");
+    ASSERT_OK(_tsfile_writer_close(writer), "close writer failed");
+    // _tsfile_writer_close already deletes the writer, no need to free
+
+    // Free the record after writing
+    _free_tsfile_ts_record(&record);
+
+    // Create reader to verify the written data
+    auto* reader = (storage::TsFileReader*)tsfile_reader_new(filename, &code);
+    ASSERT_OK(code, "create reader failed");
+
+    // Query the data we just wrote
+    char* sensors[] = {str_measurement_id, text_measurement_id,
+                       date_measurement_id};
+    auto* result = (storage::ResultSet*)_tsfile_reader_query_device(
+        reader, device_id, sensors, 3, 0, 100, &code);
+    ASSERT_OK(code, "query device failed");
+
+    // Verify the retrieved data matches what we inserted
+    bool has_next = false;
+    int row_count = 0;
+    while (!result->next(has_next) && has_next) {
```

Comment:
Rewrite `!result->next(has_next)` to make it clear that it is checking the status code.